src/command.c

```diff
@@ -244,9 +244,18 @@ bool cmd_swdp_scan(target *t, int argc, char **argv)
 static void display_target(int i, target *t, void *context)
 {
 	(void)context;
-	gdb_outf("%2d   %c  %s %s\n", i, target_attached(t)?'*':' ',
-			 target_driver_name(t),
-			 (target_core_name(t)) ? target_core_name(t): "");
+	if (!strcmp(target_driver_name(t), "ARM Cortex-M")) {
```

Comment:
Since `"ARM Cortex-M"` is used twice, could this be a define somewhere?

Reply:
Any good idea welcome. And why is hexdigits twice in memory?
> strings -n 8 blackmagic.bin | sort | uniq -c | grep -v "    1 "
      2 0123456789abcdef
      3 2F;F!F(F
      2 ARM Cortex-M
      2 [FBF1F F
> grep -I 0123456789abcdef -r
hex_utils.c:static const char hexdigits[] = "0123456789abcdef";

Reply:
good question.. pretty weird. Poking around the symbols it **_looks_** like it's coming from `<ctype.h>`. perhaps from the include in `gdb_main.c`? Looks like it was added somewhat recently. Consider eliminating one or the other, or utilizing `ctype.h` in `hex_utils.c`?

As for sharing around constant strings, you could coalesce the common strings into a shared module to reduce the amount of duplication. If you wanted to go far with it, there are 49 instances of `STM32*` that could be reduced to one (replaced with "%s" and an extra bit of time in a formatting method, so ~144 bytes of savings), or 7 instances of `Black Magic*` for 54 bytes.

Really not sure if any target is strapped for flash though... just thinking out loud.